aiogram/types/update.py

```diff
@@ -53,3 +53,62 @@ class Update(TelegramObject):
     """*Optional*. The bot's chat member status was updated in a chat. For private chats, this update is received only when the bot is blocked or unblocked by the user."""
     chat_member: Optional[ChatMemberUpdated] = None
     """*Optional*. A chat member's status was updated in a chat. The bot must be an administrator in the chat and must explicitly specify 'chat_member' in the list of *allowed_updates* to receive these updates."""
+
+    @property
+    def event(self) -> Tuple[str, TelegramObject]:
+        """
+        Detect content type
+
+        Return update type and event
+        If update type unknown raise UpdateTypeLookupError
+
+        :return:
+        """
+        event: TelegramObject
+        if self.message:
+            update_type = "message"
+            event = self.message
+        elif self.edited_message:
+            update_type = "edited_message"
+            event = self.edited_message
+        elif self.channel_post:
+            update_type = "channel_post"
+            event = self.channel_post
+        elif self.edited_channel_post:
+            update_type = "edited_channel_post"
+            event = self.edited_channel_post
+        elif self.inline_query:
+            update_type = "inline_query"
+            event = self.inline_query
+        elif self.chosen_inline_result:
+            update_type = "chosen_inline_result"
+            event = self.chosen_inline_result
+        elif self.callback_query:
+            update_type = "callback_query"
+            event = self.callback_query
+        elif self.shipping_query:
+            update_type = "shipping_query"
+            event = self.shipping_query
+        elif self.pre_checkout_query:
+            update_type = "pre_checkout_query"
+            event = self.pre_checkout_query
+        elif self.poll:
+            update_type = "poll"
+            event = self.poll
+        elif self.poll_answer:
+            update_type = "poll_answer"
+            event = self.poll_answer
+        elif self.my_chat_member:
+            update_type = "my_chat_member"
+            event = self.my_chat_member
+        elif self.chat_member:
+            update_type = "chat_member"
+            event = self.chat_member
+        else:
+            raise UpdateTypeLookupError("Unknown update type")
+
+        return update_type, event
```

Comment:
Many `elif` seems redundant here.
You can you either use `if - return` (look at 2.x), either `for` loop with `getattr` (depends on speed of execution)

Reply:
I tried to use the following implementation but it was slower than the current one

```python3
 def event(self) -> (str, TelegramObject):
        """
        Return update type and update content
        :return:
        """
        update_fields_names = update.__fields_set__

        update_type = None
        update_event = None

        for field_name in update_fields_names:
            field = getattr(update, field_name)
            if isinstance(field, TelegramObject):
                update_type = field_name
                update_event = field
                break

        if not update_event:
            UpdateTypeLookupError("Unknown update type")
        return update_type, update_event
```

![image](https://user-images.githubusercontent.com/58224121/130313366-f09b6c9b-d76e-46e7-85b4-ed0759b2c4ca.png)

Reply:
Just a quick note on this example. It can minified to something like:
```python3
 def event(self) -> (str, TelegramObject):
        update_fields_names = self.__fields_set__

        for field_name in update_fields_names:
            field = getattr(update, field_name)
            if isinstance(field, TelegramObject):
                return field_name, field

        raise UpdateTypeLookupError("Update does not contain any known event type.")
```

Reply:
```python3
print(self.__fields_set__)
{'update_id', 'message'}
```

```
Call times: 1000000
Update type last     Loop     0:00:00.556002
Update type last     If       0:00:00.626001
--------------------------------------------------------
Update type first    Loop     0:00:00.610567
Update type first    If       0:00:00.220004
--------------------------------------------------------
Call times: 1000000
Update type last     Loop     0:00:00.617021
Update type last     If       0:00:00.623003
--------------------------------------------------------
Update type first    Loop     0:00:00.613993
Update type first    If       0:00:00.217004
--------------------------------------------------------
Call times: 1000000
Update type last     Loop     0:00:00.627014
Update type last     If       0:00:00.633000
--------------------------------------------------------
Update type first    Loop     0:00:00.614973
Update type first    If       0:00:00.218997
--------------------------------------------------------
```